streamz/dataframe/core.py

```diff
@@ -245,14 +245,16 @@ def __setitem__(self, key, value):
         else:
             example = self.example.copy()
             example[key] = value
-            result = self.map_partitions(pd.DataFrame.assign, self, **{key: value})
+            df_type = type(self.example)
+            result = self.map_partitions(df_type.assign, self, **{key: value})
 
         self.stream = result.stream
         self.example = result.example
         return self
 
     def query(self, expr, **kwargs):
-        return self.map_partitions(pd.DataFrame.query, self, expr, **kwargs)
+        df_type = type(self.example)
+        return self.map_partitions(df_type.query, self, expr, **kwargs)
```

Comment:
This is fine, but if this happens more often you may want to use something like

```python
from dask.utils import M
return self.map_partitions(M.query, ...)
```

See https://github.com/dask/dask/blob/967e30c3a4b17400b91b83e0fc28dd1da50c06b9/dask/utils.py#L700

Reply:
I would leave it as is for now though, and only use the methodcaller object if this starts to happen more frequently.

Reply:
Sure, will refer to methodcaller if I have to do this more often.